pkg/controller/machine/controller.go

```diff
@@ -339,9 +339,10 @@ func (r *ReconcileMachine) drainNode(machine *machinev1.Machine) error {
 			klog.Info(fmt.Sprintf("%s pod from Node", verbStr),
 				"pod", fmt.Sprintf("%s/%s", pod.Name, pod.Namespace))
 		},
-		Out:    writer{klog.Info},
-		ErrOut: writer{klog.Error},
-		DryRun: false,
+		Out:                             writer{klog.Info},
+		ErrOut:                          writer{klog.Error},
+		DryRun:                          false,
+		SkipWaitForDeleteTimeoutSeconds: 60 * 5,
```

Comment:
may be move this value to a constant?
Do we may be want to set this value only when the node is unreachable and log so?

Reply:
cc @bison @JoelSpeed @michaelgugino wdyt

Reply:
I would agree this should be a constant. As for only setting when a node is unreachable, I think it will be quite a rare case where pods are sticking around for that long because of shut down reasons, but I have seen it (pods holding long lived connections draining over 10 mins to reduce reconnection spikes).

So I think if we can detect that the node is unreachable, then only setting this if the node is unreachable would be better

Reply:
Can looking at `status.conditions` be a good check for an unreachable node?

Reply:
That or the taints placed on the node, whichever is easiest. This might help determine when to say it's unreachable https://github.com/kubernetes/kubernetes/blob/ed3cc6afea6fa3d9f8e2a1544daaa12f87d2b65c/pkg/controller/nodelifecycle/node_lifecycle_controller.go#L69-L104

To me that suggests the `NodeReady` condition being in status `Unknown` is the only "unreachable" state, do you agree?

Reply:
Yes, you are right here https://kubernetes.io/docs/concepts/configuration/taint-and-toleration/#taint-based-evictions

Reply:
Upstream related PR https://github.com/kubernetes-sigs/cluster-api/pull/2165